### PR TITLE
[WIP] Move linting/test/flow to root package

### PR DESCRIPTION
**Pull request by skovhus**
This moves linting, testing and flow to the root package. This is a fairly standard pattern when maintaining a mono-repo. 

Besides this I fixed some linting issues and now enforce that linting is checked by CI.

Closes https://github.com/codemix/flow-runtime/issues/189

### CODE REVIEW

**Comment by jedwards1211** [comment]
@skovhus nice work, though I'm nervous about how big a change moving these packages to the root is.

It's also a problem that the `prepublish` scripts don't run `eslint`, `flow`, and `test`.  (and `prepublish` is deprecated in favor of `prepublishOnly`.)

What would you think about having the `prepublishOnly` of each package run `eslint`, `flow`, `test`, and `build`, and then making the root `ci` script run the `prepublishOnly` of each package?  I could make that change.

**Comment by gajus** [comment]
Instead of using `prepublish`, consider using git hooks.

See husky as an example of automated git hooks, https://github.com/typicode/husky.

**Comment by skovhus** [comment]
Let us close this due to inactivity. 

Introducing husky would be a good idea. :)

**Comment by gajus** [comment]
> Let us close this due to inactivity.

Is there a particular reason to closing it?

I am happy to merge the PR if we resolve the existing conflicts.

**Comment by skovhus** [comment]
> Is there a particular reason to closing it?

Unfortunately I don't really have time to get this PR landed right now. But removing 36K lines of code would be nice.

Feel free to cherry-pick my commits. : )

**Comment by jedwards1211** [comment]
@skovhus in the course of trying to upgrade the build to babel 7 I've noticed differences in the test configuration between the various pages, especially how `packages/flow-runtime/test-polyfill.js` passes babel config directly but the same file in most of the other packages don't.  Can you summarize how you accommodated differences like this in root-level config?  (There are a lot of changes to look through)

**Comment by skovhus** [comment]
I would avoid different test configurations. 

But if you really need it for some reason, the test files itself can include a polyfill file with an import/require statement.
